source/game.gamescriptexpression.bmx

```diff
@@ -344,7 +345,11 @@ End Function
 
 
 Function _EvaluateProgrammeDataCast:SToken(data:TProgrammeData, params:STokenGroup Var, tokenOffset:int) 'inline
-	Local castIndex:Int = params.GetToken(2 + tokenOffset).valueLong
+	If Not params.HasToken(1 + tokenOffset, ETokenValueType.Integer)
+		Return New SToken( TK_ERROR, "No valid cast number passed", params.GetToken(0) )
```

Comment:
Ich weiß noch nicht so recht, wofür der letzte Parameter des Error-Tokens verwendet wird. Hilfreich für die Analyse wäre ja die Ausgabe des gesamten Ausdrucks, nicht nur der Fehlermeldung...

Reply:
Anhand des Tokens wird die "Zeilen/Cursor"-Angabe gemacht (wo ist der Fehler passiert - bzw "wo fing er an"). In unserem Fall also der Beginn der kompletten hier evaluierten Expression.
Es handelt sich hier ja nicht um eine "Subexpression" (nicht immer)

Reply:
Das wird nur gebraucht, wenn man "reveal()" nutzt (also "debuggt" bzw halt wirklich den Error-Text aufbohren woellte)

also "token.Reveal()" oder "params.Reveal()"